src/figuresClasses.ts

```diff
@@ -1,19 +1,100 @@
-export interface Figure {
+type Shapes = 'triangle' | 'circle' | 'rectangle';
+type Colors = 'red' | 'green' | 'blue';
 
+export interface Figure {
+  shape: Shapes;
+  color: Colors;
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: Shapes;
 
-}
+  a: number;
+
+  b: number;
+
+  c: number;
+
+  constructor(public color: Colors, a: number, b: number, c: number) {
+    this.shape = 'triangle';
+    this.a = a;
+    this.b = b;
+    this.c = c;
+
+    this.isTriangle();
+  }
+
+  getArea(): number {
+    const semiPerim = (this.a + this.b + this.c) / 2;
+
+    const area = Math.sqrt(
+      semiPerim
+      * (semiPerim - this.a) * (semiPerim - this.b) * (semiPerim - this.c),
+    );
+
+    return +area.toFixed(2);
+  }
 
-export class Circle {
+  isTriangle(): void {
+    if ((this.a === 0 || this.b === 0 || this.c === 0)) {
```

Comment:
```suggestion
    if (!this.a || !this.b || !this.c) {
```

Reply:
But better to check if it's less or equal to 0 in that case to prevent negative values